### ISSUE TITLE
Has Okta dependency

### ISSUE THREAD

**Issue by PenelopeFudd** [lyft/cartography#459]
**Bug Template**

*Title*: *ModuleNotFoundError: No module named 'okta.framework'*

*Description*:
> I've just installed Cartography and I'm running it for the first time, and it dies with an error about not finding the okta.framework module.

*To Reproduce*:
> Install Neo4j (brew cask install neo4j)
> Start Neo4j
> Set up Python3 virtual environment (virtualenv -p $(which python3) python; source python/bin/activate; pip install --upgrade pip)
> Install cartography (pip install cartography)
> Run cartography: 
```
cartography --neo4j-uri bolt://localhost:7687
```

*Logs*:
> If applicable, copy and paste your console log with the failing stack trace.
```
$ cartography --neo4j-uri bolt://localhost:7687
Traceback (most recent call last):                                                                                                                                                      
  File "/Users/me/work/cartography/python/bin/cartography", line 5, in <module>
    from cartography.cli import main
  File "/Users/me/work/cartography/python/lib/python3.8/site-packages/cartography/cli.py", line 7, in <module>
    import cartography.sync
  File "/Users/me/work/cartography/python/lib/python3.8/site-packages/cartography/sync.py", line 16, in <module>
    import cartography.intel.okta
  File "/Users/me/work/cartography/python/lib/python3.8/site-packages/cartography/intel/okta/__init__.py", line 3, in <module>
    from okta.framework.OktaError import OktaError
ModuleNotFoundError: No module named 'okta.framework'
```

*Screenshots*:
> If applicable, add screenshots to help explain your problem.

*Please complete the following information:*:
>  - Cartography release version or commit hash: cartography-0.28.0
>  - Python version: 3.8.5
>  - OS (feel free to omit this if you don't think it's relevant to your issue): OSX Catalina 10.15.7

*Additional context*:
> Our company does use Okta, but not for AWS.

**Comment by achantavy** [comment]
Hi, sorry for the trouble. See #424 and #425 for more details about this problem.

As a temporary work-around, you can try running `pip install okta==0.0.4` in your environment and then trying to run cartography again.

We fixed this issue in our 0.29.0 release, but we've been having trouble getting that  pushed to pypi because of (yet another) dependency problem.

Sorry again for the trouble, there's lots of different moving pieces that we're fixing at the moment! 😅

**Comment by PenelopeFudd** [comment]
Ok!  I apologize for opening an issue for this, I searched for 'okta' issues but didn't see any descriptions that matched in the open issues.

**Comment by PenelopeFudd** [comment]
Next error:
```
$ cartography --neo4j-uri bolt://127.0.0.1:7687
Traceback (most recent call last):
  File "/Users/me/work/cartography/python/bin/cartography", line 8, in <module>
    sys.exit(main())
  File "/Users/me/work/cartography/python/lib/python3.8/site-packages/cartography/cli.py", line 348, in main
    return CLI(default_sync, prog='cartography').main(argv)
  File "/Users/me/work/cartography/python/lib/python3.8/site-packages/cartography/cli.py", line 328, in main
    return cartography.sync.run_with_config(self.sync, config)
  File "/Users/me/work/cartography/python/lib/python3.8/site-packages/cartography/sync.py", line 106, in run_with_config
    neo4j_driver = GraphDatabase.driver(
  File "/Users/me/work/cartography/python/lib/python3.8/site-packages/neo4j/__init__.py", line 120, in driver
    return Driver(uri, **config)
  File "/Users/me/work/cartography/python/lib/python3.8/site-packages/neo4j/__init__.py", line 161, in __new__
    return subclass(uri, **config)
  File "/Users/me/work/cartography/python/lib/python3.8/site-packages/neo4j/__init__.py", line 235, in __new__
    pool.release(pool.acquire())
  File "/Users/me/work/cartography/python/lib/python3.8/site-packages/neobolt/direct.py", line 715, in acquire
    return self.acquire_direct(self.address)
  File "/Users/me/work/cartography/python/lib/python3.8/site-packages/neobolt/direct.py", line 608, in acquire_direct
    connection = self.connector(address, error_handler=self.connection_error_handler)
  File "/Users/me/work/cartography/python/lib/python3.8/site-packages/neo4j/__init__.py", line 232, in connector
    return connect(address, **dict(config, **kwargs))
  File "/Users/me/work/cartography/python/lib/python3.8/site-packages/neobolt/direct.py", line 972, in connect
    raise last_error
  File "/Users/me/work/cartography/python/lib/python3.8/site-packages/neobolt/direct.py", line 963, in connect
    s, der_encoded_server_certificate = _secure(s, host, security_plan.ssl_context, **config)
  File "/Users/me/work/cartography/python/lib/python3.8/site-packages/neobolt/direct.py", line 854, in _secure
    s = ssl_context.wrap_socket(s, server_hostname=host if HAS_SNI and host else None)
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/ssl.py", line 1040, in _create
    self.do_handshake()
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/ssl.py", line 1309, in do_handshake
    self._sslobj.do_handshake()
OSError: [Errno 0] Error
```
Isn't the zero error = "success"?

**Comment by PenelopeFudd** [comment]
Apologies again; I should just get the 0.29.0 version from github.  Sorry!

**Comment by achantavy** [comment]
Double check that you have the Neo4j 3.5.19 database (not 4.x!), and the neo4j 1.7.6 python driver.

**Comment by PenelopeFudd** [comment]
It's `Neo4j Desktop Version 1.3.10 (1.3.10.36)`, which is whatever version `brew cask install neo4j` decided to download.

**Comment by PenelopeFudd** [comment]
Apparently there's a `neo4j` formula and a `neo4j` cask.  I'm downloading the formula now.  Sigh.

**Comment by PenelopeFudd** [comment]
Aaaaand apparently Brew is downloading neo4j-community-4.1.3-unix.tar.gz :-(

**Comment by achantavy** [comment]
Haha yeah, you'll want to follow these steps: https://github.com/lyft/cartography/blob/master/docs/setup/install.md#cartography-installation

In case you get stuck on that, I put together a setup video for devs of Cartography, but a lot of the steps probably apply to users too: https://www.youtube.com/watch?v=4-Ps7c_BmjE

**Comment by PenelopeFudd** [comment]
From watching the video, TIL `pip install -e .`  😄   

Also, Neo4j.com has redesigned their website, making it harder to find Neo4j 3.5.x.  👎🏻 

Ok, it seems to be running now! ... and running... and running... Hmm, needs a progress bar.  :-)

Thanks!

**Comment by achantavy** [comment]
Alright, I released 0.29.1 and that should be in Pypi now which should not be affected by the okta.framework issue so you don't need to run an editable install anymore unless you want to be running tests and deving cartography :-)

**Comment by PenelopeFudd** [comment]
Thanks!


On November 13, 2020 11:44:43 a.m. PST, Alex Chantavy <notifications@github.com> wrote:
>Alright, I released 0.29.1 and that should be in Pypi now which should not be affected by the okta.framework issue so you don't need to run an editable install anymore unless you want to be running tests and deving cartography :-)
>
>-- 
>You are receiving this because you modified the open/close state.
>Reply to this email directly or view it on GitHub:
>https://github.com/lyft/cartography/issues/459#issuecomment-726996271
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.
